### PR TITLE
fix(auction): publish updates when lockedPrice changes

### LINKED ISSUE #7567: AuctionBook isn't publishing every update to locked price
## What is the Problem Being Solved?

@samsiegart reports that when the UI follows `book0.startPrice`, it doesn't see the locked quote changing as expected.

## Description of the Design

in `auctionBook.js`, both `lockOraclePriceFroRound()` and `endAuction()` should call `publishBookData()`.

## Security Considerations

none.

## Scaling Considerations

none.

## Test Plan

update some tests that monitor changing published book data.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
closes: #7567

## Description

Add calls to `publishBookData()` when the lockedPrice is changed.

### Security Considerations

None

### Scaling Considerations

No impact

### Documentation Considerations

N/A

### Testing Considerations

Updated existing tests.

### CODE REVIEW

File: packages/inter-protocol/src/auction/auctionBook.js [hunk 1]
```diff
@@ -612,9 +612,12 @@ export const prepareAuctionBook = (baggage, zcf, makeRecorderKit) => {
           return scaledBidBook.hasOrders() || priceBook.hasOrders();
         },
         lockOraclePriceForRound() {
-          const { updatingOracleQuote } = this.state;
+          const { facets, state } = this;
+          const { updatingOracleQuote } = state;
```
Comment (turadg) [inline packages/inter-protocol/src/auction/auctionBook.js]: consider deeper destructuring,
```suggestion
          const { facets: { helper }, state: { updatingOracleQuote } } = this;
```
fwiw, now that you have `state`, line 619 could be shorter

**Review by turadg — APPROVED** [review]

File: packages/inter-protocol/src/auction/auctionBook.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/auction/auctionBook.js]: I tried the deeper destructuring, and prettier turned it into 4 lines, so I reverted.

> fwiw, now that you have state, line 619 could be shorter

Oh, yeah. Definitely an omission.

**Comment by samsiegart** [comment]
I'm not seeing a locked price or book currently with 2 minutes until liquidation (and `PriceLockPeriod = 300`):

```
$ agd query vstorage keys published.auction
children:
- governance
- schedule
pagination: null
```

image

I only see `book0` show up after liquidation has started. I'd expect it to show up at least as early as the price lock period.

**Comment by Chris-Hibbert** [comment]
>  I only see book0 show up after liquidation has started. I'd expect it to show up at least as early as the price lock period.

I don't know what that means. I don't know when nodes in the graph show up. I wouldn't expect the nodes to go away, though. What happens on later rounds of the auction? I presume the `book0` node sticks around, and the price updates should be visible at the appropriate times.

**Comment by samsiegart** [comment]
I mean `published.auction.book0` exists only after liquidation starts, but we need to read it before liquidation starts to know if the vault is going to be liquidated. I'd have to jump through some more hoops I'm unfamiliar with to test later rounds, but already this seems undesired behavior. I'm concerned about the `startPrice` which shouldn't change throughout rounds anyway.

**Comment by Chris-Hibbert** [comment]
>  `published.auction.book0` exists only after liquidation starts, but we need to read it before liquidation starts to know if the vault is going to be liquidated.

@turadg, 

When are the various nodes in the tree created? What can the publisher do to cause them to appear earlier?

**Comment by turadg** [comment]
> When are the various nodes in the tree created?

`makeChildNode` creates the StorageNode but it doesn't write to RPC until a `setValue`. IIUC that will write data into the block which it was called.

> What can the publisher do to cause them to appear earlier?

The `PublishKit` and `RecorderKit` abstractions call setValue. So publish/write with one of them should do it.

**Review by turadg — APPROVED** [review]

**Comment by Chris-Hibbert** [comment]
@sam, I changed the way it schedules locking of prices. This should address the problem you are seeing.

**Review by samsiegart — APPROVED** [review]
Looks like we have a happy UI now:
image

**Comment by Chris-Hibbert** [comment]
@mergify refresh

**Pull request merged**